### PR TITLE
Patch broken import for private io modules

### PR DESCRIPTION
**Pull request by nabobalis**
This should make the modules actually importable now.

Fixes https://github.com/sunpy/sunpy/issues/6373

### CODE REVIEW

**Comment by jgieseler** [comment]
> Why does moving these imports fix the issue?

It's not the moving, but adding a dot before the underscore: from `._`cdf import *

**Comment by nabobalis** [comment]
I added some unit tests to check importlib can import the old modules.

**Review by dstansby — CHANGES_REQUESTED** [review]
Instead of blanked ignoring the warning in the test config, the warnings should be caught or filtered in the new tests, to make sure no-one accidentally uses the deprecated API anywhere else in the package.

**Comment by nabobalis** [comment]
```
ERROR sunpy/io/cdf.py - sunpy.util.exceptions.SunpyDeprecationWarning: The `sunpy.io.cdf` module is deprecated, as it was designed for internal use.
ERROR sunpy/io/file_tools.py - sunpy.util.exceptions.SunpyDeprecationWarning: The `sunpy.io.file_tools` module is deprecated, as it was designed for internal use.
ERROR sunpy/io/jp2.py - sunpy.util.exceptions.SunpyDeprecationWarning: The `sunpy.io.jp2` module is deprecated, as it was designed for internal use.
```
My understanding is that pytest is importing the Python files and that triggers the warnings causing the test suite to fail. I do not know how to deal with that without using the ignore section of setup.cfg.

File: setup.cfg [hunk 1]
```diff
@@ -224,6 +224,8 @@ filterwarnings =
     # https://github.com/h5py/h5py/issues/2243 for upstream issues about this warning being
     # emitted
     ignore:`product` is deprecated as of NumPy 1.25.0:DeprecationWarning
+    # Depecreated for 5.0 - IO modules
+    ignore:.*module is deprecated, as it was designed for internal use
```
[the remark this answers is not in the document]
Comment (wtbarnes) [reply setup.cfg:228]: I'm not comfortable with it, but I don't have a better suggestion.

**Review by wtbarnes — APPROVED** [review]

**Review by dstansby — CHANGES_REQUESTED** [review]
I think I've found a way to avoid having the global warning ignores: https://github.com/nabobalis/sunpy/pull/1

**Review by dstansby — APPROVED** [review]
👍 if CI passes. The devdeps figure test fail looks un-related

**Pull request merged**